integration-tests/opentelemetry-reactive/src/main/java/io/quarkus/it/opentelemetry/reactive/ReactiveResource.java

```diff
@@ -0,0 +1,37 @@
+package io.quarkus.it.opentelemetry.reactive;
+
+import javax.inject.Inject;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.QueryParam;
+
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.Tracer;
+import io.smallrye.mutiny.Uni;
+
+@Path("/reactive")
+public class ReactiveResource {
+    @Inject
+    Tracer tracer;
+
+    @GET
+    public Uni<String> helloGet(@QueryParam("name") String name) {
+        Span span = tracer.spanBuilder("helloGet").startSpan();
+        try {
+            return Uni.createFrom().item("Hello " + name);
```

Comment:
This is not async. You should try with something leaving the event loop with be coming back. 

As a result you would not be able to use the try /finally.

Reply:
Yes, I know. This was enough to reproduce the problem since it would still execute the rest call in the resteasy reactive context. I guess I can write a more realistic test.